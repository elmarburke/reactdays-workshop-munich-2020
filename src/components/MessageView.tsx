import React from "react";
import styled from "styled-components";
import { Message } from "../domain";

const Article = styled.article`
  background-color: pink;
  border: 1px solid green;
  color: blueviolet;
`;

interface Props {
  message: Message;
}

const MessageView: React.FunctionComponent<Props> = ({ message }) => {
  return (
    <Article>
      {message.message}

      <footer>{message.author}</footer>
    </Article>
  );
};

export default MessageView;
