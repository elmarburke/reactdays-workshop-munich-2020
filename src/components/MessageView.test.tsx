import React from "react";
import { render } from "@testing-library/react";
import { screen } from "@testing-library/dom";
import MessageView from "./MessageView";

describe("<MessageView />", () => {
  it("shows the message", async () => {
    render(
      <MessageView
        message={{
          id: "1",
          message: "Hello World",
          author: "Hans Wurst",
          date: 1,
        }}
      />
    );

    expect(screen.getByText("Hello World")).toMatchSnapshot();
  });
});
