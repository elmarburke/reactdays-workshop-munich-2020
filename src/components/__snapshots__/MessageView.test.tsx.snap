// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageView /> shows the message 1`] = `
<article
  class="sc-AxjAm dcMTwH"
>
  Hello World
  <footer>
    Hans Wurst
  </footer>
</article>
`;
